### PR TITLE
Set up Account Details

### PR DESCRIPTION
**Pull request by thejoefriel**
#### Issues this PR relates to
#51 
#52 
#131 

=======

_NOTE: REMEMBER NOT TO REVIEW UNLESS ALL ACCEPTANCE CRITERIA HAS BEEN TICKED OFF_

---

#### Any info the reviewer needs to know to test 
e.g. Add env vars, run npm install

---

#### Any info the team needs to consider before merged into development
e.g. Will a database need to be rebuilt (and is this ok if existing data), add env vars

---

#### Quick Pull Request Checklist

Please make sure you ensure the following has been done before marking your PR as awaiting-review. For full checklist and process review, please refer to Process Guide (#8)

- Branch is up to date with develop
- Project runs
- You can download and install locally, and run the project locally
- All existing and new features run without bugs
- Checked that features/design works across all agreed browsers + devices
- The new work has been checked with Axe accessibility checker and doesn't throw significant errors  
- Linter is used
- Code is consistent with agreed team style
- Naming of files is consistent with project 
- Tests all pass
- No console logs

### CODE REVIEW

File: server/modules/user/use-cases/update-account.js [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+        profileImageMedia = await Media.createMedia({
+          fileName: name,
+          fileType,
+          size,
+          key,
+          bucket,
+          bucketRegion,
+          createdBy: user.id,
+        });
+        // console.log(`profileImageMedia`, profileImageMedia);
```
Comment (RamyAlshurafa) [inline server/modules/user/use-cases/update-account.js:53]:
```suggestion
```

**Review by RamyAlshurafa — CHANGES_REQUESTED** [review]
- [ ] on therapist account page,
  -  i uploaded an image and submited, it worked fine
  - then edited the name, and hit submit again and the back end threw an error,
```
[1] error: duplicate key value violates unique constraint "media_key_key"
[1]     at Connection.parseE (/home/ramy/chiltern-music-therapy/node_modules/pg/lib/connection.js:581:48)
[1]     at Connection.parseMessage (/home/ramy/chiltern-music-therapy/node_modules/pg/lib/connection.js:380:19)
[1]     at Socket.<anonymous> (/home/ramy/chiltern-music-therapy/node_modules/pg/lib/connection.js:116:22)
[1]     at Socket.emit (events.js:314:20)
[1]     at Socket.EventEmitter.emit (domain.js:506:15)
[1]     at addChunk (_stream_readable.js:297:12)
[1]     at readableAddChunk (_stream_readable.js:272:9)
[1]     at Socket.Readable.push (_stream_readable.js:213:10)
[1]     at TCP.onStreamRead (internal/stream_base_commons.js:188:23)
[1]     at TCP.callbackTrampoline (internal/async_hooks.js:126:14) {
[1]   length: 248,
[1]   severity: 'ERROR',
[1]   code: '23505',
[1]   detail: 'Key (key)=(4/image/jpeg-fffb3e69-5144-4c07-b049-7779ae108c23-COW_0197.JPG) already exists.',
[1]   hint: undefined,
[1]   position: undefined,
[1]   internalPosition: undefined,
[1]   internalQuery: undefined,
[1]   where: undefined,
[1]   schema: 'public',
[1]   table: 'media',
[1]   column: undefined,
[1]   dataType: undefined,
[1]   constraint: 'media_key_key',
[1]   file: 'nbtinsert.c',
[1]   line: '570',
[1]   routine: '_bt_check_unique'
[1] }
[1] POST /api/users/update-account 500 15.668 ms - 96
```

- [ ] when just updated any field other than the image, i got this error
```
[1] TypeError: Cannot read property 'id' of undefined
[1]     at _callee$ (/home/ramy/chiltern-music-therapy/server/modules/user/use-cases/update-account.js:64:48)
```

File: client/src/pages/Client/MyAccount/index.js [hunk 2]
```diff
@@ -0,0 +1,175 @@
...
+    }
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [accountDetails]);
+
+  useEffect(() => {
+    if (user.id) {
+      const { firstName, lastName, email } = user;
+      setAccountDetails({
+        firstName: firstName[0],
+        lastName: lastName[0],
```
Comment (RamyAlshurafa) [inline client/src/pages/Client/MyAccount/index.js:103]: i think you should show the full `firstName` and `lastName` or just store the first letter in db?

**Review by RamyAlshurafa — CHANGES_REQUESTED** [review]

File: client/src/pages/Client/MyAccount/index.js (hunk 2 above)
Comment (thejoefriel) [reply client/src/pages/Client/MyAccount/index.js:103]: Yeah, I'll change add client so we only store the first letter anyway

File: client/src/api-calls/users.js [hunk 3]
```diff
@@ -103,6 +103,30 @@ const getTherapists = async ({ options } = {}) => {
   }
 };
 
+const updateAccount = async (accountData) => {
+  try {
+    const { data } = await axios.post(
+      `${USERS_BASE}/update-account`,
+      accountData
```
[the remark this answers is not in the document]
Comment (thejoefriel) [reply client/src/api-calls/users.js:110]: done

File: client/src/pages/Client/MyAccount/index.js [hunk 4]
```diff
@@ -0,0 +1,175 @@
...
+      handleUpdate();
+    }
+  };
+
+  useEffect(() => {
+    if (updateAttempt) {
+      validateForm();
+    }
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [accountDetails]);
```
[the remark this answers is not in the document]
Comment (thejoefriel) [reply client/src/pages/Client/MyAccount/index.js:96]: done

File: client/src/pages/Client/MyAccount/index.js (hunk 2 above)
Comment (thejoefriel) [reply client/src/pages/Client/MyAccount/index.js:103]: done

**Review by RamyAlshurafa — APPROVED** [review]
merging this and opening new issue to fix the profile image and refactor get profile image url endpoint

**Pull request merged**